scripts/lib/mkcloud-ipmi.sh

```diff
@@ -0,0 +1,45 @@
+function ipmi_cleanup
+{
+    for node in $(seq $nodenumber) ; do
+        if ip link show v${cloud}${node}-x >/dev/null ; then
+            ip link del dev v${cloud}${node}-x
+        fi
+        screen -X -S $cloud-node$node-bmc-lan quit
+    done
+}
+
+function mkveth
+{
+    local nodenumber=$1
+    local bmc_addr=$2
+    local iface=v$cloud$nodenumber
+    ip link add dev $iface-x type veth peer name $iface-y
+    ip link set dev $iface-x up
+    ip link set dev $iface-y up
+    ip addr add ${bmc_addr}/${adminnetmask} dev $iface-x
+    brctl addif $cloudbr $iface-y
+}
+
+function generate_lan_config()
+{
+    local nodenumber=$1
+    local user=$2
+    local password=$3
+    local listenaddr=$4
+    local nodename=$cloud-node$nodenumber
+    local lanconfig=/tmp/ipmi-lan-$nodename.conf
```

Comment:
I would be more concerned about using /tmp in combination with mktemp because of the possibility that things don't get cleaned up and /tmp fills up. By using a consistent name we can only have as many temp files as nodes that are running, so it won't grow exponentially.